missing dependency on Compat_digest

The automated builds on [build.opensuse.org](https://build.opensuse.org/package/show/Archiving:unison:next/unison) currently fail to build, as of commit 20251016T091732 (7516f58). The last successful build was 20251013T133013 (197467a):
```
[   38s] Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.io5Mty
[   38s] + umask 022
[   38s] + cd /home/abuild/rpmbuild/BUILD
[   38s] + /usr/bin/rm -rf /home/abuild/rpmbuild/BUILDROOT/unison-20251016T091732.7516f58-Aun150700.115.1.x86_64
[   38s] ++ dirname /home/abuild/rpmbuild/BUILDROOT/unison-20251016T091732.7516f58-Aun150700.115.1.x86_64
[   38s] + /usr/bin/mkdir -p /home/abuild/rpmbuild/BUILDROOT
[   38s] + /usr/bin/mkdir /home/abuild/rpmbuild/BUILDROOT/unison-20251016T091732.7516f58-Aun150700.115.1.x86_64
[   38s] + cd unison-20251016T091732.7516f58
[   38s] + /usr/bin/make -O -j4 PREFIX=/usr
[   38s] cd src && /usr/bin/make
[   39s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] Not on macOS. macOS native GUI will not be built.
[   39s] 
[   39s] Building for Unix
[   39s] NATIVE = true
[   39s] 
[   39s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] /usr/bin/make -f Makefile.OCaml all
[   39s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] ocamlopt: ubase/rx.mli ---> ubase/rx.cmi
[   39s] ocamlopt -g -I lwt -I ubase -I system -I system/generic -I lwt/generic -open Compat_digest -I +unix -I +str    -c /home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src/ubase/rx.mli
[   39s] File "command line", line 1:
[   39s] Error: Unbound module Compat_digest
[   39s] make[2]: *** [Makefile.OCaml:195: ubase/rx.cmi] Error 2
[   39s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] make[2]: *** Waiting for unfinished jobs....
[   39s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] ocamlopt: ubase/umarshal.mli ---> ubase/umarshal.cmi
[   39s] ocamlopt -g -I lwt -I ubase -I system -I system/generic -I lwt/generic -open Compat_digest -I +unix -I +str    -c /home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src/ubase/umarshal.mli
[   39s] File "command line", line 1:
[   39s] Error: Unbound module Compat_digest
[   39s] make[2]: *** [Makefile.OCaml:195: ubase/umarshal.cmi] Error 2
[   39s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] ocamlopt: unicode_tables.ml ---> unicode_tables.cmx
[   39s] ocamlopt -g -I lwt -I ubase -I system -I system/generic -I lwt/generic -open Compat_digest -I +unix -I +str    -c /home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src/unicode_tables.ml
[   39s] File "command line", line 1:
[   39s] Error: Unbound module Compat_digest
[   39s] make[2]: *** [Makefile.OCaml:203: unicode_tables.cmx] Error 2
[   39s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] ocamlopt -I +unix -I +str -c compat_digest.ml
[   39s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/unison-20251016T091732.7516f58/src'
[   39s] make[1]: *** [Makefile:42: all] Error 2
[   39s] make: *** [Makefile:22: src] Error 2
```

--- a/src/Makefile.OCaml
+++ b/src/Makefile.OCaml
@@ -53,7 +53,6 @@ clean::
 # File extensions will be substituted for the native code version
 
 OCAMLOBJS = \
-          $(COMPAT_OBJS) \
           ubase/umarshal.cmo \
           ubase/rx.cmo \
           \
@@ -155,6 +154,8 @@ lwt/lwt_unix.cmx: lwt/$(SYSTEM)/lwt_unix_impl.cmx
 lwt/win/lwt_win.cmo: lwt/win/lwt_unix_impl.cmo system/system_win.cmo
 lwt/win/lwt_win.cmx: lwt/win/lwt_unix_impl.cmx system/system_win.cmx
 
+$(CAMLOBJS) $(CAMLOBJS_FSM) $(OCAMLOBJS:.cmo=.cmi) system.cmi fspath.cmi fs.cmi system.cmo fspath.cmo fs.cmo system.cmx fspath.cmx fs.cmx lwt/lwt_unix.cmi lwt/lwt_unix.cmo lwt/lwt_unix.cmx lwt/win/lwt_win.cmi lwt/win/lwt_win.cmo lwt/win/lwt_win.cmx: $(COMPAT_OBJS)
+
 .PHONY: depend
 depend:
 	ocamlc -depend $(DEP_INCLFLAGS) *.mli *.ml */*.ml */*.mli */*/*.ml */*/*.mli > .depend
@@ -210,15 +211,15 @@ CAMLCFLAGS_X = $(CAMLCFLAGS) $(CAMLCFLAGS_FSM_X)
 # are expanded at parsing, so we need all variables to be defined by then.
 include Makefile2.cfg
 
-$(NAME)$(EXEC_EXT): $(CAMLOBJS) $(CAMLOBJS_TUI) $(COBJS)
+$(NAME)$(EXEC_EXT): $(COMPAT_OBJS) $(CAMLOBJS) $(CAMLOBJS_TUI) $(COBJS)
 	@echo Linking $@
 	$(CAMLC) -verbose $(CAMLFLAGS_X) $(CAMLLDFLAGS) -o $@ $(CAMLLIBS) $(ALL__SRC) $(CLIBS)
 
-$(NAME_GUI)$(EXEC_EXT): $(CAMLOBJS) $(CAMLOBJS_GUI) $(COBJS)
+$(NAME_GUI)$(EXEC_EXT): $(COMPAT_OBJS) $(CAMLOBJS) $(CAMLOBJS_GUI) $(COBJS)
 	@echo Linking $@
 	$(CAMLC) -verbose $(CAMLFLAGS_X) $(CAMLLDFLAGS) $(CAMLLDFLAGS_GUI) -o $@ $(CAMLLIBS) $(CAMLLIBS_GUI) $(ALL__SRC) $(CLIBS)
 
-$(NAME_FSM)$(EXEC_EXT): $(CAMLOBJS_FSM) $(FSMCOBJS)
+$(NAME_FSM)$(EXEC_EXT): $(COMPAT_OBJS) $(CAMLOBJS_FSM) $(FSMCOBJS)
 	@echo Linking $@
 	$(CAMLC) -verbose $(CAMLFLAGS_X) $(CAMLLDFLAGS) -o $@ $(CAMLLIBS_FSM) $(ALL__SRC) $(CLIBS) $(CLIBS_FSM)
 
@@ -228,9 +229,9 @@ $(NAME_FSM)$(EXEC_EXT): $(CAMLOBJS_FSM) $(FSMCOBJS)
 # -output-complete-obj, available since OCaml 4.02.2. Still, this option can't
 # be used here because there is a bug which breaks linking on macOS when using
 # threads (needed by uimacbridge): https://github.com/ocaml/ocaml/issues/7933
-$(NAME)-blob.o: $(CAMLOBJS) $(CAMLOBJS_MAC) $(COBJS)
+$(NAME)-blob.o: $(COMPAT_OBJS) $(CAMLOBJS) $(CAMLOBJS_MAC) $(COBJS)
 	@echo Linking $@
-	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS_X) -o u-b.o $(CAMLLDFLAGS) $(CAMLLIBS) $(CAMLLIBS_MAC) $(CLIBS) $(CAMLOBJS) $(CAMLOBJS_MAC)
+	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS_X) -o u-b.o $(CAMLLDFLAGS) $(CAMLLIBS) $(CAMLLIBS_MAC) $(CLIBS) $(COMPAT_OBJS) $(CAMLOBJS) $(CAMLOBJS_MAC)
 	$(LD) -r -keep_private_externs -o $@ u-b.o $(COBJS)
 	$(RM) u-b.o
 

--- a/src/make_tools.ml
+++ b/src/make_tools.ml
@@ -255,7 +255,7 @@ let () =
   |> List.filter (fun (_, cond) -> cond)
   |> List.iter (fun (name, _) ->
       "CAMLFLAGS" <-+= "-open " ^ (String.capitalize_ascii name);
-      "COMPAT_OBJS" <-+= name ^ ".cmo";
+      "COMPAT_OBJS" <-+= name ^ "." ^ (if native then "cmx" else "cmo");
       outp (Printf.sprintf
         "%s.cmo: %s.ml\n\
         \t$(OCAMLC) $(OCAMLINCLUDES) -c $(ALL__SRC)\n\

